m4/check_pkg_cuda.m4

```diff
@@ -21,40 +22,67 @@ AC_DEFUN([CHECK_PKG_CUDA], [
         [],
         [test "${with_cuda}" = "no"],
         [check_pkg_found=no],
-        [AS_IF([test -d ${with_cuda}/lib64], [check_pkg_libdir="lib64"], [check_pkg_libdir="lib"])
+        [AS_IF([test -d $(realpath ${with_cuda})/lib64], [check_pkg_libdir="lib64"], [check_pkg_libdir="lib"])
+         CUDA_LDFLAGS="-L$(realpath ${with_cuda})/${check_pkg_libdir}"
+         CUDA_LIBS="-lcudart_static -lrt -ldl"
+         CUDA_CPPFLAGS="-I$(realpath $(realpath ${with_cuda})/include)"
+         CUDA_CFLAGS="-fPIC"
```

Comment:
Is there a place where you're finding you need -fPIC?  It should already be there for the library, and you shouldn't need it for the binaries.